cmake/onnxruntime_providers.cmake

```diff
@@ -851,7 +851,7 @@ if (onnxruntime_USE_RKNPU)
   onnxruntime_add_include_to_target(onnxruntime_providers_rknpu onnxruntime_common onnxruntime_framework onnx onnx_proto protobuf::libprotobuf-lite flatbuffers)
   target_link_libraries(onnxruntime_providers_rknpu PRIVATE -lrknpu_ddk)
   add_dependencies(onnxruntime_providers_rknpu onnx ${onnxruntime_EXTERNAL_DEPENDENCIES})
-  set_target_properties(onnxruntime_providers_rknpu PROPERTIES CXX_STANDARD 14)
+  set_target_properties(onnxruntime_providers_rknpu PROPERTIES CXX_STANDARD 17)
```

Comment:
do we need to explicitly set this if CMAKE_CXX_STANDARD is set to the same value already?